opsdroid/parsers/witai.py

```diff
@@ -0,0 +1,76 @@
+"""A helper function for parsing and executing wit.ai skills."""
+
+import logging
+import json
+
+import aiohttp
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def call_witai(message, config):
+    """Call the wit.ai api and return the response."""
+    async with aiohttp.ClientSession() as session:
+        headers = {
+            "Authorization": "Bearer " + config['access-token']
+        }
+        payload = {
+            "v": "20170307",
+            "q": message.text
+        }
+        resp = await session.get("https://api.wit.ai/message?v={}&q={}".format(
+            payload['v'], payload['q']), headers=headers)
+        result = await resp.json()
+        _LOGGER.debug("wit.ai response - " + json.dumps(result))
+        _LOGGER.info(result)
+        return result
+
+
+async def parse_witai(opsdroid, message, config):
+    """Parse a message against all witai skills."""
+    # pylint: disable=broad-except
+    # We want to catch all exceptions coming from a skill module and not
+    # halt the application. If a skill throws an exception it just doesn't
+    # give a response to the user, so an error response should be given.
+    if 'access-token' in config:
+        try:
+            result = await call_witai(message, config)
+        except aiohttp.ClientOSError:
+            _LOGGER.error("No response from wit.ai, check your network.")
+            return
+
+        if 'code' in result:
+            _LOGGER.error("wit.ai error - " + str(result['code'])
+                          + " " + str(result['error']))
+            return
+        elif result['entities'] == {}:
+            _LOGGER.error("wit.ai error - No intent found. Did you "
+                          "forget to create one?")
+            return
+
+        confidence = result['entities']['intent'][0]['confidence']
```

Comment:
If the string doesn't match an intent then there will be no intent paramater in the response resulting in an error. This needs to be caught.

```python
Traceback (most recent call last):
  File "/opt/boxen/homebrew/Cellar/python3/3.5.2_3/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/Users/jacob/Projects/opsdroid/opsdroid/opsdroid/parsers/witai.py", line 51, in parse_witai
    confidence = result['entities']['intent'][0]['confidence']
KeyError: 'intent'
```